libs/components/tabs/src/lib/modules/tabs/tab-button.component.ts

```diff
@@ -45,14 +56,65 @@ export class SkyTabButtonComponent {
   public disabled: boolean;
 
   @Input()
-  public tabStyle: SkyTabsetStyle;
+  public tabIndex: SkyTabIndex;
+
+  @Input()
+  public get tabStyle(): SkyTabsetStyle {
+    return this.#_tabStyle;
+  }
+
+  public set tabStyle(style: SkyTabsetStyle) {
+    this.#_tabStyle = style;
+    this.elementRole = style === 'tabs' ? 'tab' : undefined;
+  }
 
   @Output()
   public buttonClick = new EventEmitter<void>();
 
   @Output()
   public closeClick = new EventEmitter<void>();
 
+  constructor(
+    elementRef: ElementRef,
+    adapterService: SkyTabButtonAdapterService,
+    changeDetectorRef: ChangeDetectorRef,
+    tabsetService: SkyTabsetService
+  ) {
+    this.#adapterService = adapterService;
+    this.#changeDetectorRef = changeDetectorRef;
+    this.#elementRef = elementRef;
+    this.#tabsetService = tabsetService;
+  }
+
+  public elementRole: string | undefined = 'tab';
```

Comment:
I mentioned this below, but we should store default values (that are used more than once) in a const at the top of the page. `const DEFAULT_ELEMENT_ROLE = 'tab'`.